q2_feature_table/_summarize/_visualizer.py

```diff
@@ -32,20 +32,47 @@
 
 def tabulate_seqs(output_dir: str, data: DNAIterator,
                   taxonomy: pd.DataFrame = None,
-                  metadata: qiime2.Metadata = None) -> None:
-    sequences = []
+                  metadata: qiime2.Metadata = None,
+                  merge_method: str = 'strict') -> None:
+
+    display_sequences = set()
+    sequences = {}
     seq_lengths = []
     with open(os.path.join(output_dir, 'sequences.fasta'), 'w') as fh:
         for sequence in data:
             skbio.io.write(sequence, format='fasta', into=fh)
             str_seq = str(sequence)
             seq_len = len(str_seq)
-            sequences.append({'id': sequence.metadata['id'],
-                              'len': seq_len,
-                              'url': _blast_url_template % str_seq,
-                              'seq': str_seq})
+            display_sequences.add(sequence.metadata['id'])
+            sequences[sequence.metadata['id']]\
+                = {'len': seq_len,
+                   'url': _blast_url_template % str_seq,
+                   'seq': str_seq}
             seq_lengths.append(seq_len)
 
+    if metadata is not None:
+        metadata_df = metadata.to_dataframe()
+        if merge_method == 'union':
+            display_sequences = display_sequences.union(metadata_df.index)
+        elif merge_method == 'intersect':
+            display_sequences = display_sequences.intersection(
+                metadata_df.index)
+        elif merge_method == 'strict':
+            if set(metadata_df.index) != display_sequences:
+                raise Exception('Merge method is strict and IDs do not match')
+    if taxonomy is not None:
+        for name in taxonomy.keys():
+            if merge_method == 'union':
+                display_sequences = display_sequences.union(
+                    taxonomy[name][0].index)
+            elif merge_method == 'intersect':
+                display_sequences = display_sequences.intersection(
+                    taxonomy[name][0].index)
+            elif merge_method == 'strict':
+                if set(taxonomy[name][0].index) != display_sequences:
+                    raise Exception('Merge method is strict and IDs do not \
+                        match')
```

Comment:
Something like

```
for value in taxonomy.values()
```

then value is a dataframe so you can just do

```
value.index
```

everywhere you get the index

Reply:
That doesn't work, taxonomy.values returns a numpy representation of the same data, I tried that already, but the dataframe will always have one row of values at index zero, unless the user somehow made a mistake.